File: app/src/main/java/alex/com/githubchecker/components/github/activities/PRDiffActivity.java
package alex.com.githubchecker.components.github.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import javax.inject.Inject;

import alex.com.githubchecker.components.app.BaseActivity;
import alex.com.githubchecker.components.app.GithubCheckerApp;
import alex.com.githubchecker.components.github.pullrequest.PRDiffPresenter;
import alex.com.githubchecker.components.github.pullrequest.PRDiffView;
import alex.com.githubchecker.components.github.repo.GithubModel;
import butterknife.ButterKnife;

/**
 * Created by Alex on 11/17/2017.
 */

public class PRDiffActivity extends BaseActivity {

    private static String KEY_PR_ID = "id";
    PRDiffView view;
    PRDiffPresenter presenter;
    @Inject GithubModel githubModel;

    public static void Show(Context context, Integer pullRequestId) {
        Intent in = new Intent(context, PRDiffActivity.class);
        in.putExtra(KEY_PR_ID, pullRequestId);
        context.startActivity(in);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Integer pullRequestId = getIntent().getIntExtra(KEY_PR_ID, 0);

        ButterKnife.bind(this);
        GithubCheckerApp.getGithubComponent().inject(this);

        view = new PRDiffView(this);
        presenter = new PRDiffPresenter(githubModel, view, pullRequestId);

        setContentView(view.view());
        presenter.onCreate();
    }
}

File: app/src/main/java/alex/com/githubchecker/components/github/pullrequest/PRDiffActivity.kt
package alex.com.githubchecker.components.github.pullrequest

import alex.com.githubchecker.components.app.BaseActivity
import alex.com.githubchecker.components.app.GithubCheckerApp
import alex.com.githubchecker.models.dagger.GithubModel
import android.content.Context
import android.content.Intent
import android.os.Bundle
import javax.inject.Inject

/**
 * Created by Alex on 11/17/2017.
 */

class PRDiffActivity : BaseActivity() {

    companion object {

        private val KEY_PR_ID = "id"

        fun Show(context: Context, pullRequestId: Int?) {
            val intent = Intent(context, PRDiffActivity::class.java)
            intent.putExtra(KEY_PR_ID, pullRequestId)
            context.startActivity(intent)
        }
    }

    private lateinit var view: PRDiffView
    private lateinit var presenter: PRDiffPresenter

    @Inject
    lateinit var githubModel: GithubModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        GithubCheckerApp.githubComponent.inject(this)

        view = PRDiffView(this)

        val pullRequestId = intent.getIntExtra(KEY_PR_ID, 0)
        presenter = PRDiffPresenter(githubModel, view, pullRequestId)

        presenter.onCreate()
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.onDestroy()
    }
}